java-sdk/README.asciidoc

```diff
@@ -0,0 +1,212 @@
+Similarly, you can generate a Fusion-compatible JSON schema by implementing `Model`.
+
+You can see an example of a configuration implementation
+https://github.com/lucidworks/connectors-sdk-resources/blob/master/java-sdk/connectors/random-connector/src/main/java/com/lucidworks/fusion/connector/plugin/RandomContentConfig.java[here^].
+
+More detailed information on the configuration and schema capabilities can be found 
+ifdef::env-github[]
+link:configuration.asciidoc[here].
+endif::[]
+ifndef::env-github[]
+link:/fusion-server/{version}/search-development/getting-data-in/connectors/sdk/java-sdk/java-connector-dev.html#configuration[here].
+endif::[]
+
+=== The Fetcher
+The `Fetcher` is where most of the work is done for any connector.
+This interface provides methods that define how data is fetched and indexed.
+
+For a connector to have its content indexed into Fusion, it must emit messages, which are simple objects
+that contain metadata and content. See the link:#message-definitions[message definitions] below for more details.
+
+==== Lifecycle
+
+The `Fetcher` has a simple lifecycle for dealing with fetching content.
+The flow outline is as follows:
+```
+    start()           - called once when the Job is started
+        preFetch()    - called for each Phase
+            fetch()   - called for each input to fetch
+        postFetch()   - called for each Phase
+    stop()            - called once when the Job is stopped
+```
+
+====== start()
+When a `ConnectorJob` starts, the Fetcher's `start()` method is called. The main use case for a start() call
+is to run setup code. This method is called once per job run.
+
+The result of a start() call is metadata, which tells the connector controller service how to perform a crawl. Currently,
+the main control feature enabled is to define fetch "phases". Each phase encapsulates a `preFetch() -> fetch() -> postFetch()`
+call chain, and is defined by setting an ordered list of phase names in the start() response. This allows the connector
+to define a chain of distinct fetch blocks within a single job.
+
+====== preFetch()
+This method is called once per fetch phase and allows a connector to emit data, that will eventually be
+sent to the fetch() method. A common use case for this is to emit a large amount of metadata, which the controller service
+later distributes across the cluster to perform the real fetch work.
+
+====== fetch()
+This is the primary fetch method for a crawler-like connector (for example, a file system or web connector). Messages emitted here
+can include `Documents`, which are indexed directly into the associated content collection. There are several other
+types of messages; see link:#message-definitions[Message Definitions].
+
+====== postFetch()
+Called after each phase for a given job run.
+
+====== stop()
+Called once per job run, at the end.
+
+==== Phases
```

Comment:
Phases are now slightly different as well. We'll probably need to get a Jira in for someone to make sure all of this is updated / accurate.

Reply:
Actually, it looks like this is the main file that needs the most updating. We'll try to help with this as a team throughout the week.